### ISSUE TITLE
 test accuracy increase slow in IID Cifar10.

### ISSUE THREAD

**Issue by Mangata1** [TsingZ0/PFL-Non-IID#13]
Can you provide the latest code，many thanks.

**Comment by TsingZ0** [comment]
Hi, @Mangata1.

This is already the latest code. 

Can you provide more details on this issue?

**Comment by Mangata1** [comment]
Hello, @TsingZ0, I'm glad to get your reply!
I am using your code to train a federated learning model on IID cifar10 dataset.  The local model I am using is resnet. 
I found that no matter how I set the learning rate and hyperparameters like local eporch, the test accuracy remained around 10%.
To test your code, I tried using generate_cifar10.py to generate a collection of devices with only one client. I only use this one client to run FedAvg. You know this is equivalent to using resnet to train a centralized model on the cifar10 dataset. Even so, the test accuracy remains around 10%. At the same time, I observed that the training accuracy has reached 100%.
No matter how I modify the hyperparameters, the test accuracy is still very low.
The following figures are the parameters I set and the results(Don't care about the learning rate setting, I tried other learning rate values).
![屏幕截图 2022-06-26 001243](https://user-images.githubusercontent.com/50471866/175782073-da7dc0ff-9a50-49d3-9f7b-fc4a2a6350c2.png)
![屏幕截图 2022-06-26 001355](https://user-images.githubusercontent.com/50471866/175782085-61b7fd5c-44e8-4239-b185-c83ae8c34f15.png)
 I found in other people's issues that this phenomenon also exists in other datasets. I guess there may be a problem with the data generation of iid.
Thanks again for your reply and your code sharing.

**Comment by TsingZ0** [comment]
Hi, @Mangata1 

I checked the code again, and address this issue by setting `shuffle=True` in `./dataset/utils/dataset_utils.py`. 
Besides, I also update other codes in `./dataset/`. Please check it. 

Thank you for your help!

Here is an example for Cifar10 in the IID setting with resnet18:

Data generation: `python generate_cifar10.py iid - -`
Simulation: `python -u main.py -lr 0.1 -lbs 16 -nc 20 -jr 1 -nb 10 -data Cifar10 -m resnet -algo FedAvg -gr 2000 -did 0`
![image](https://user-images.githubusercontent.com/24207672/175812288-1b0f7e0e-b8de-4bc6-88f2-6a96a13b1930.png)

**Comment by Mangata1** [comment]
Hello, @TsingZ0
Good Job！
Thanks for sharing the new code.

**Comment by oneheartyx** [comment]
你好请问为什么我使用用图片上的参数，在round number 3时测试精度只有0.2276？（与0.4233相比差距还是很大的）
感谢你们开源了这么好的代码
